### PR TITLE
Skip down/kill/rm when DockerComposeRule#skipShutdown is true.

### PR DESCRIPTION
**Pull request by jkozlowski**

### CODE REVIEW

**Comment by iamdanfox** [comment]
:+1:

File: src/main/java/com/palantir/docker/compose/DockerComposeRule.java [hunk 1]
```diff
@@ -99,10 +104,17 @@ public void before() throws IOException, InterruptedException {
     @Override
     public void after() {
         try {
-            log.debug("Killing docker-compose cluster");
-            dockerCompose().down();
-            dockerCompose().kill();
-            dockerCompose().rm();
+            if (skipShutdown()) {
+                log.error("******************************************************************************************\n"
+                        + "******** Skipping docker-compose down/kill/rm; This should not happen on CI! *************\n"
```
Comment (joelea) [inline src/main/java/com/palantir/docker/compose/DockerComposeRule.java]: Can you make it clear that the rule has been configured this way, it's not some weird docker error.

File: README.md [hunk 2]
```diff
@@ -149,6 +149,25 @@ public class DockerCompositionTest {
...
 
+Skipping shutdown
+-----------------
+
+To skip shutdown of containers after tests are finished executing:
+
+```java
+public class DockerCompositionTest {
+
+    private static final boolean SKIP_SHUTDOWN = true;
```
Comment (joelea) [inline README.md]: This is probably less clear than just using the boolean in this case.

File: src/main/java/com/palantir/docker/compose/DockerComposeRule.java (hunk 1 above)
Comment (iamdanfox) [inline src/main/java/com/palantir/docker/compose/DockerComposeRule.java]: I actually don't like the `This should not happen on CI!` message.

Comment (jkozlowski) [inline src/main/java/com/palantir/docker/compose/DockerComposeRule.java]: How about (with error still):

> docker-compose-rule has been configured to skip docker-compose shutdown: 
> This means the containers will be left running after tests finish executing. 
> If you see this message when running on CI it means you are potentially abandoning
> long running processes.

File: README.md (hunk 2 above)
Comment (jkozlowski) [inline README.md]: Cool, will update to just call skipShutdown(true), is that what you mean?

File: src/main/java/com/palantir/docker/compose/DockerComposeRule.java (hunk 1 above)
Comment (jkozlowski) [inline src/main/java/com/palantir/docker/compose/DockerComposeRule.java]: Updated the PR.

File: README.md (hunk 2 above)
Comment (jkozlowski) [inline README.md]: Updated PR.

**Comment by joelea** [comment]
LGTM, will merge and release when builds pass.

**Comment by jkozlowski** [comment]
Sweet, thanks!

**Pull request merged**